R/calculate_silhouette_width.R

```diff
@@ -0,0 +1,78 @@
+#' Calculate silhouette width scores from an integrated SCE object.
+#'
+#' This function performs replicated calculations on downsampled data.
+#'
+#' @param integrated_sce The integrated SCE object
+#' @param pc_name The name that allows access to the PCs. Example: fastMNN_PCA
+#' @param batch_column The variable in `integrated_sce` indicating the grouping of interest.
+#'  Generally this is either batches or cell types. Default is "library_id".
+#' @param frac_cells The fraction of cells to downsample to. Default: 0.8
+#' @param nreps The number of times to repeat sub-sampling procedure. Default: 20
+#' @param seed Seed for initializing random sampling
+#'
+#' @return Tibble with five columns: `rep`, representing the given downsampling replicate;
+#'   `silhouette_width`, the calculated silhouette width for the given `rep`; `silhouette_cluster`,
+#'   the assigned cluster for the cell during silhouette calculation, i.e. the true identity;
+#'   `other_cluster`, the other assigned for the cell during silhouette calculation;
+#'   `pc_name`, the name associated with the pc results
+#'
+#' @import SingleCellExperiment
```

Comment:
This doesn't really matter as it is imported elsewhere in the package but it is good practice to add it where it is used: 
```suggestion
#' @import SingleCellExperiment
#' @importFrom rlang .data
```